woocommerce/woocommerce
# Add range operator in Remote Inbox Notification

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #44787 

This PR adds `range` operation in Remote Inbox Notification to support testing a number's range. This PR also updates Remote Spec Validation schema files to support `range`



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Run unit tests and make sure they pass. `./vendor/bin/phpunit --testsuite shard2`
2. Install and activate WooCommerce Beta Tester
3. Go to `Tools -> WCA Test Helper -> Remote Spec Rule Validator`
4. Get the value of `woocommerce_remote_variant_assignment` option. 
5. Paste the following rule and adjust `value` for testing.

```
[
    {
        "type": "option",
        "option_name": "woocommerce_remote_variant_assignment",
        "value": [0, 10],
        "operation": "range"
    }
]
```



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Support range operator in Remote Inbox Notification 

#### Comment 

</details>
